tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -39,6 +45,96 @@ private void getCapabilities(PrintWriter out, HttpServletRequest hsreq) {
 		gcdw.finishXML();
 	}
 	
+	/**
+	 * Checks request parameters for errors.
+	 * Will send back an XML Exception if any errors are encountered.
+	 * 
+	 * @param request parameter value
+	 * @param version parameter value
+	 * @param service parameter value
+	 * @return an ExceptionWriter if any errors occurred or null if none occurred
+	 */
+	private WFSExceptionWriter checkParametersForError(String request, String version, String service) {

```

Comment:
Good parameter validation throughout this method! When the validation finds that users have submitted invalid values, it might be helpful to include the invalid values in the WFS Exception responses. However this pull request is looking really good, so that isn't a blocker for merging.